darshan-runtime/lib/darshan-core.c

```diff
@@ -394,12 +407,37 @@ void darshan_core_initialize(int argc, char **argv)
         darshan_core_fprintf(stderr, "darshan:init\t%d\t%f\n", nprocs, init_time);
     }
 
+
+#ifdef HAVE_LDMS
```

Comment:
I think this entire block could just be shifted into `darshan_ldms_connector_initialize()`, with you passing whatever is needed as arguments? That would tidy up the integration a bit more.